src/netlist/module.cpp

```diff
@@ -236,23 +236,24 @@ std::set<std::shared_ptr<net>> module::get_input_nets() const
 {
     std::set<std::shared_ptr<net>> res;
     auto gates = get_gates(nullptr, true);
-    std::set<std::shared_ptr<net>> seen;
     for (const auto& gate : gates)
     {
         for (const auto& net : gate->get_fan_in_nets())
         {
-            if (seen.find(net) != seen.end())
-            {
-                continue;
-            }
-            seen.insert(net);
```

Comment:
Why remove this check?
If we decide to keep it, it should be changed to an `std::unordered_set<u32>` and keep track of net ids though...

Reply:
I basically modified this method to use the same approach as in `module::get_output_nets()` since that method already implements checking for multiple endpoints. In that method, _you_ removed the check in https://github.com/emsec/hal/commit/a0395c618e187f37f60506b41a057d420a4ab157 so I wanted to be consistent.

Technically, since the data is inserted into a set, this check is not required for correctness. If we get a key duplicate on `net`, it's not inserted into `res` since `std::set` keys are unique. There might be a performance implication though for big modules, e.g. if we encounter the clock net a thousand times. That could be a good reason to keep the check in place. What do you think?

Reply:
The check in `get_output_nets` was superfluous before the multi-driven nets, since all output nets were module-unique by definition, so I removed it.
This did not apply to input nets though!
Now, with multi-driven nets, this optimization could actually benefit both, input and output nets.
However, input nets that occur multiple times are still way more common than comparable output nets.
Please add the optimization to both functions.